src/Event/TimesheetUpdateEvent.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Kimai time-tracking app.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Event;
+
+use App\Entity\Timesheet;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * This event can be used to call functions after timesheet creations/updates
+ */
+final class TimesheetUpdateEvent extends Event
```

Comment:
Instead of one event I would like to have one class for every use-case (TimesheetUpdateEvent, TimesheetDeleteEvent, TimesheetMultiUpdateEvent, TimesheetMultiDeletEvent...), with two base classes AbstractTimesheetEvent and AbstractTimesheetMultiEvent.
Then the method signatures could be made explicit: either for one entity or for a collection.